backend/src/federation/index.test.ts

```diff
@@ -0,0 +1,162 @@
+  from: {
+    id: Buffer.from('somone'),
+    host: '188.95.53.5',
+    port: 63561,
+  },
+  to: { id: null, host: '83.53.31.27', port: 55723 },
+  peers: [
+    {
+      publicKey: Buffer.from('some-public-key'),
+      relayAddresses: [],
+    },
+  ],
+}
+
+const nodeLookupMock = jest.fn().mockResolvedValue([lookupResultMock])
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const socketEventMocks: { [key: string]: any } = {}
+
+const socketOnMock = jest.fn().mockImplementation((key: string, callback) => {
+  socketEventMocks[key] = callback
+})
+
+const nodeConnectMock = jest.fn().mockImplementation(() => {
+  return {
+    on: socketOnMock,
+    once: socketOnMock,
+  }
+})
+
+DHT.hash.mockImplementation(() => {
+  return Buffer.from(TEST_TOPIC)
+})
+
+DHT.keyPair.mockImplementation(() => {
+  return keyPairMock
+})
+
+DHT.mockImplementation(() => {
+  return {
+    createServer: nodeCreateServerMock,
+    announce: nodeAnnounceMock,
+    lookup: nodeLookupMock,
+    connect: nodeConnectMock,
+  }
+})
+
+describe('federation', () => {
+  describe('call startDHT', () => {
+    const hashSpy = jest.spyOn(DHT, 'hash')
+    const keyPairSpy = jest.spyOn(DHT, 'keyPair')
+
+    beforeEach(async () => {
+      DHT.mockClear()
+      jest.clearAllMocks()
+      await startDHT(TEST_TOPIC)
+    })
+
+    it('calls DHT.hash', () => {
+      expect(hashSpy).toBeCalledWith(Buffer.from(TEST_TOPIC))
+    })
+
+    it('creates a key pair', () => {
+      expect(keyPairSpy).toBeCalledWith(expect.any(Buffer))
+    })
+
+    it('initializes a new DHT object', () => {
+      expect(DHT).toBeCalledWith({ keyPair: keyPairMock })
+    })
+
+    describe('DHT node', () => {
+      it('creates a server', () => {
+        expect(nodeCreateServerMock).toBeCalled()
+      })
+
+      it('listens on the server', () => {
+        expect(serverListenSpy).toBeCalled()
+      })
+
+      describe('timers', () => {
+        beforeEach(() => {
+          jest.runOnlyPendingTimers()
+        })
+
+        it('announces on topic', () => {
+          expect(nodeAnnounceMock).toBeCalledWith(Buffer.from(TEST_TOPIC), keyPairMock)
+        })
+
+        it('looks up on topic', () => {
+          expect(nodeLookupMock).toBeCalledWith(Buffer.from(TEST_TOPIC))
+        })
+      })
+
+      describe('server connection event', () => {
+        beforeEach(() => {
+          serverEventMocks.connection({
+            remotePublicKey: Buffer.from('another-public-key'),
+            on: socketOnMock,
+          })
+        })
+
+        it('can be triggered', () => {
+          expect(socketOnMock).toBeCalled()
+        })
+
+        describe('socket events', () => {
+          describe('on data', () => {
+            beforeEach(() => {
+              socketEventMocks.data(Buffer.from('some-data'))
+            })
+
+            it('can be triggered', () => {
+              expect(true).toBe(true)
+            })
+          })
+        })
```

Comment:
I think this is the crucial part where you recieve data and we handle it. We would have several test scenarios where we recieve valid or invalid data - this should be the main test case.

Will this work like above?

`expect(socketOnMocks[data]).toBeCalledWith(Buffer.from('some-data'))`